misc/dataframeclustering/src/processors/dataframeclustering.cpp

```diff
@@ -0,0 +1,181 @@
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <inviwo/dataframeclustering/processors/dataframeclustering.h>
+#include <inviwo/dataframeclustering/dataframeclusteringmodule.h>
+#include <modules/python3/pybindutils.h>
+
+namespace inviwo {
+
+// The Class Identifier has to be globally unique. Use a reverse DNS naming
+// scheme
+const ProcessorInfo DataFrameClustering::processorInfo_{
+    "org.inviwo.DataFrameClustering",  // Class identifier
+    "Data Frame Clustering",           // Display name
+    "Clustering",                      // Category
+    CodeState::Experimental,           // Code state
+    {{Tag::PY, Tag::CPU}}              // Tags
+};
+const ProcessorInfo DataFrameClustering::getProcessorInfo() const { return processorInfo_; }
+
+DataFrameClustering::DataFrameClustering(InviwoApplication *app)
+    : Processor()
+    , dataFrame_("dataFrame")
+    , newDataFrame_("newDataFrame")
+
+    , method_("method", "Method",
+              {{"kmeans", "KMeans"},
+               {"dbscan", "DBSCAN"},
+               {"agglo", "Agglomerative"},
+               {"spectral", "Spectral"}})
+
+    , numberOfClusters_("numberOfClusters", "Number Of Clusters", 3,
+                        {1, ConstraintBehavior::Immutable}, {10, ConstraintBehavior::Ignore})
+
+    , kmeans_("kmeans_", "K-Means")
+    , dbscan_("dbscan_", "DBSCAN")
+    , agglomerative_("agglomerative_", "Agglomerative")
+    , spectral_("spectral", "Spectral")
+
+    , columnName_("columnName", "Column Name", "Cluster")
+
+    , eps_("eps", "Epsilon", 0.3f, {0.001f, ConstraintBehavior::Ignore},
+           {1.f, ConstraintBehavior::Ignore}, 0.001f)
+    , N_("N", "Min Points", 10, {1, ConstraintBehavior::Immutable},
+         {100, ConstraintBehavior::Ignore})
+
+    , numberOfFoundClusters_("numberOfFoundClusters", "NumberOfFoundClusters", 0,
+                             {0, ConstraintBehavior::Immutable}, {1000, ConstraintBehavior::Ignore},
+                             1, InvalidationLevel::Valid, PropertySemantics::Text)
+
+    , linkage_("linkage", "Linkage",
+               {{"single", "Single"},
+                {"complete", "complete"},
+                {"average", "average"},
+                {"weighted", "weighted"},
+                {"centroid", "centroid"},
+                {"median", "median"},
+                {"ward", "ward"}})
+
+    , columns_("columns", "Columns")
+    , script_(app->getModuleByType<DataFrameClusteringModule>()->getPath(ModulePath::Scripts) +
+              "/dataframeclustering.py") {
```

Comment:
script missing in PR

Reply:
Wiered, I copy-pasted the whole folder and did 'git add . '